hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -963,7 +965,15 @@ protected RequestFactory createRequestFactory() {
     // Any encoding type
     String contentEncoding = getConf().getTrimmed(CONTENT_ENCODING, null);
 
-    String storageClass = getConf().getTrimmed(STORAGE_CLASS, null);
+    StorageClass storageClass;
+    try {
+      String storageClassConf = getConf()
+          .getTrimmed(STORAGE_CLASS, "")
+          .toUpperCase(Locale.US);
+      storageClass = StorageClass.fromValue(storageClassConf);
+    } catch (IllegalArgumentException e) {
+      storageClass = null;
```

Comment:
good q here. i wonder what to do. I would recommend logging at info/warn on the unknown value and then continuing. so the try/catch would only surround line 973 and log the value of storageClassConf in the error.

Reply:
sounds good, i will add some logging at warn to let users know that they should check storage class config